drivers/bms_ic/bq769x2/bq769x2.c

```diff
@@ -590,7 +590,7 @@ static int bms_ic_bq769x2_configure(const struct device *dev, struct bms_ic_conf
         return -EIO;
     }
 
-    return (flags == actual_flags) ? 0 : -EINVAL;
+    return actual_flags;
```

Comment:
Thought about this before. But let's imagine a chip wich has no voltage regulator (but supports configuring voltage limits):

| Passed flags | Original PR | Suggested change |
|----|-----|-----|
| `BMS_IC_CONF_VOLTAGE_REGS` | `0` | `-EINVAL` |
| `BMS_IC_CONF_VOLTAGE_LIMITS \| BMS_IC_CONF_VOLTAGE_REGS` | `BMS_IC_CONF_VOLTAGE_LIMITS` | `BMS_IC_CONF_VOLTAGE_LIMITS` |

I think that would be somewhat inconsistent.

The idea behind this API change is that you can indicate serious errors (like communication failures), but that you don't get a failure if you attempt to configure something that the chip doesn't support. Instead the application can compare the returned flags with the passed flags to figure out if all configurations are actually supported.

I don't currently see another alternative how to indicate all these different possible outcomes. (except for adding another parameter with a pointer to a flags variable which will be used as the output)

Reply:
You're correct, however, the original PR presents an inconsistency where a `0` signals an error occurred while setting up just one configuration.
Considering your suggestion to introduce an additional parameter that uses a pointer to a flags variable for output, could this indeed be the best solution?

Reply:
True, the current proposal would also not be fully consistent.

I have updated the commit to use your original suggestion, but with `-ENOTSUP` as a return code if *none* of the requested flags could be configured. This is the usual Zephyr return code in case the hardware doesn't support a particular driver API function. `0` will never be returned anymore, just if one calls the function with flags set to `0` which would not make sense anyway.